### PR TITLE
Feature/authorization

### PR DESCRIPTION
**Pull request by Redjik**

### CODE REVIEW

File: api/app/Extensions/Controller/AuthorizesRequestsTrait.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+     *
+     * @param  string  $ability
+     * @param  array  $arguments
+     * @return HttpException
+     */
+    protected function createGateUnauthorizedException($ability, $arguments)
+    {
+        return new HttpException(403, 'This action is unauthorized.');
+    }
+}
```
Comment (zakhenry) [inline api/app/Extensions/Controller/AuthorizesRequestsTrait.php]: Why do we need all the extended functionality?

Comment (Redjik) [inline api/app/Extensions/Controller/AuthorizesRequestsTrait.php]: what do you mean? nothing extra

Comment (zakhenry) [inline api/app/Extensions/Controller/AuthorizesRequestsTrait.php]: I meant the whole file - I now see that lumen hasn't included the traits - I've added a ticket on the lumen github repo.

**Comment by zakhenry** [comment]
@Redjik I've had a look into the issue with the Request facade resolution, and the hack that you've done in the articleMeta changelog test actually hides a bigger issue - the user resolver middleware that was removed has meant that the controllers can no longer run `Request $request->user()` to get the user object. example:

```
public function putMany(Request $request, $id)
    {
        $parent = $this->findParentEntity($id);

        dd($request->user()); //returns null now (it used to get the user)

```

This also breaks the `Request::` facade as you found with the revisionable trait failing to find the user on the facade.

It does seem like a hack and I don't know why it is the case, but I think the middleware route is a workable option, and doing things like authentication registration in a middleware is a logic place to do it (though the jwt service registration would probably be better, it doesn't seem to work).

**Pull request merged**